dali/pipeline/operator/builtin/external_source.h

```diff
@@ -220,6 +220,8 @@ class ExternalSource : public Operator<Backend>, virtual public BatchSizeProvide
         device_id_(spec.GetArgument<int>("device_id")),
         dtype_(spec.GetArgument<DALIDataType>("dtype")),
         previous_dtype_(DALIDataType::DALI_NO_TYPE),
+        ndim_(spec.GetArgument<int>("ndim")),
+        previous_ndim_(-1),
```

Comment:
Do we really need "previous" now? We could just live with one value:
```C++
  if (spec.TryGetArgument(ndim_, "ndim"))
    DALI_ENFORCE(ndim >= 0, "Incorrect number of dimensions. Use positive values for tensors or 0 for scalars.);
  else
    ndim = -1;
```

Reply:
Got rid of the `previous`